packages/salesforcedx-utils-vscode/src/predicates/predicate.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// This file is meant to mimics the functionality of Predicate and Predicates from Google Guava.
+// Expand as necessary.
+
+export interface Predicate<T> {
+  apply(item: T): PredicateResponse;
+}
+export class PredicateResponse {
```

Comment:
This mimics the Predicate feature in Google Guava.
I've not added the various `and`, `or`, or `not` operations but those can be added later as we need them.